client/src/components/AddFieldAgent.js

```diff
@@ -0,0 +1,131 @@
+import { useState } from "react";
+
+function AddFieldAgent({ handleAdd, handleCancel }) {
+  const [firstName, setFirstName] = useState('');
+  const [middleName, setMiddleName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [dob, setDob] = useState('');
+  const [heightInInches, setHeightInInches] = useState('');
+  const [errors, setErrors] = useState([]);
+
+  let valid = true;
+
+  const handleFirstNameChange = (event) => {
+    setFirstName(event.target.value);
+  }
+
+  const handleMiddleNameChange = (event) => {
+    setMiddleName(event.target.value);
+  }
+
+  const handleLastNameChange = (event) => {
+    setLastName(event.target.value);
+  }
+
+  const handleDoBChange = (event) => {
+    setDob(event.target.value);
+  }
+
+  const handleHeightInInchesChange = (event) => {
+    setHeightInInches(event.target.value);
+  }
+
+  const checkForValidDate = (event) => {
+    let input = event.target.value;
+    if(input.charAt(4) !== '-' || input.charAt(7) !== '-'
+    || isNaN(input.substring(0,4)) || isNaN(input.substring(5,7)) || isNaN(input.substring(8,11))
+    || input.substring(0,4) === '' || input.substring(5,7) === '' || input.substring(8,11) === ''
+    || input.length === 11) {
+      if(isNaN(Date.parse(input))){
+        alert("Date must be in YYYY-MM-DD format");
+        valid = false;
+      }
+    }else{
+      valid = true;
+    }
+  }
+
+  const checkForHeightInInches = (event) => {
+    if(isNaN(event.target.value)) {
+      alert("Input must be a number.")
+      valid = false;
+    }else{
+      valid = true;
+    }
+  }
+
+  const handleSubmit = (event) => {
+    if(valid){
+      const newFieldAgent = {
+        firstName,
+        middleName,
+        lastName,
+        dob,
+        heightInInches
+      };
+
+      const init = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(newFieldAgent)
+      };
```

Comment:
The page seems to be refreshing and sending me back to the main list, no matter what happens here.